arch/risc-v/src/common/espressif/Kconfig

```diff
@@ -486,6 +493,215 @@ config ESP_MCPWM
 		Enable support for timer capture and motor control using
 		the Motor Control PWM peripheral.
 
+menu "Pulse Counter (PCNT) Configuration"
+	depends on ESP_PCNT
```

Comment:
Pulse counter should only be available on certain Espressif chips (C3 for instance does not have PCNT peripheral)

Reply:
Could you, please, provide me on which chips it should depend?

Reply:
PCNT support for available in: ESP32H2 and ESP32C6

Reply:
Thank you, I've added dependency.